portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,11 +27,14 @@
 public class DataServlet extends HttpServlet {
 
   ArrayList<String> messages = new ArrayList<String>();
-  
-  public DataServlet () {
-    messages.add("The sun is out today!");
-    messages.add("I just got new curtains.");
-    messages.add("Woo hoo!");
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("comment-area");
+    messages.add(0, comment);
```

Comment:
@megankj I'm calling add with the index because later in my code, when I used the messages array, I thought the easiest way to do it would be to print "messages[0]" instead of keeping track of how many messages there are. So if I don't add with the index, my code will just print the same message over and over again since .add will add the element to the end of the array

Reply:
Is there another / better way I should be doing this?

Reply:
I think you have a couple options:
1) In the JS, instead of printing `messages[0]`, can you print the last message in the array? (i.e. `messages[messages.length - 1]` or something)
2) In `doGet`, don't send the entire message array down, since the client is only using one value.  In larger applications you don't want to send more data to the client than is necessary for performance reasons.
3) Change your data structure to a stack, since you can pop on values.  I *think* `gson.toJson` would convert to json in the right format, but I'm not sure.
4) Leave as is, but leave a comment why you're explicitly adding to the beginning of the array.

This will become obsolete soon when you start storing these comments in the datastore instead of the server, so it's up to you!  (option 3 probably isn't work it because this ArrayList will soon be deleted). Personally, I think I would just modify the JS to get the last message instead of the first.

Reply:
Thanks for the detailed reply! The first option seemed the most efficient and simplest, so I went with that.